jax/_src/api.py

```diff
@@ -3245,6 +3246,50 @@ def try_to_block(x):
       return x
   return jax.tree_util.tree_map(try_to_block, x)
 
+def pure_callback(callback: Callable[..., Any], result_shape_dtypes: Any,
+                  *args: Any, **kwargs: Any):
+  """Calls a pure Python callback function from staged out JAX programs.
```

Comment:
```suggestion
  """Applies a functionally pure Python callable. Works under `jit`/`pmap`/etc.
```

Not sure if the suggested phrasing is ideal but I'm thinking:
1. "pure Python" isn't as clear as saying "functionally pure";
2. "staged out", while I love it, may not be as clear as saying it works with `jit`, etc", though if we had space we might want to say both;
3. I like "applies" better than "calls", esp when pure;
4. "from staged out JAX programs" could suggest it _only_ works when staged out.